app/components/admin/edition_images/uploaded_images_component.rb

```diff
@@ -4,4 +4,37 @@ class Admin::EditionImages::UploadedImagesComponent < ViewComponent::Base
   def initialize(edition:)
     @edition = edition
   end
+
+  def lead_image
+    if @edition.first_image_pulled_out? && !@edition.images.empty?
+      image_to_object @edition.images.first, 0
+    end
+  end
+
+  def document_images
+    lead_image_count = @edition.first_image_pulled_out? ? 1 : 0
+    if @edition.images.length > lead_image_count
+      @edition.images.drop(lead_image_count).map.with_index(1) { |image, index| image_to_object image, index }
+    end
+  end
+
+private
+
+  def image_to_object(image, index)
+    {
+      url: image.url,
+      preview_alt_text: index.zero? ? "Lead image" : "Image #{index}",
+      caption: image.caption.presence || "None",
+      alt_text: image.alt_text.presence || "None",
+      markdown: "[#{image.image_data.carrierwave_image}]",
+      links: links_for_image(image),
+    }
+  end
+
+  def links_for_image(image)
+    links = []
+    links << link_to("Edit details", edit_admin_edition_image_path(@edition, image), class: "govuk-link")
+    links << link_to("Delete image", confirm_destroy_admin_edition_image_path(@edition, image), class: "govuk-link gem-link--destructive")
+    links
```

Comment:
Just thinking that the links are very view specific, I'm not sure it's more readable getting extracted to the ruby class rather than staying in the view.

Reply:
I moved it to the class to avoid duplicating it in the view (between lead and document document images)
I don't think that it's an issue as this is ViewComponent class and hence directly tied to the view